packages/serverpod_serialization/test/json_serialization_test.dart

```diff
@@ -1,28 +1,177 @@
-    );
-    expect(stringifiedJson, '''
+  test(
+    'Given simple JSON data, when encoded using encode method with formatting, then output is formatted stringified JSON',
+    () {
+      var jsonData = {'a': 1, 'b': 2};
+      var stringifiedJson = SerializationManager.encode(
+        jsonData,
+        formatted: true,
+      );
+      expect(
+        stringifiedJson,
+        '''
 {
   "a": 1,
   "b": 2
-}''');
+}''',
+      );
+    },
+  );
+
+  group('Given a user object,', () {
+    late _User user;
+
+    setUp(() {
+      user = _User(name: 'John', password: '123');
+    });
+
+    test(
+      'when encoded using encodeForProtocol method, then password is excluded from the output',
+      () {
+        var stringifiedJson = SerializationManager.encodeForProtocol(user);
+        expect(stringifiedJson.contains('password'), isFalse);
+      },
+    );
+
+    test(
+      'when a list of user objects is encoded using encodeForProtocol method, then passwords are excluded from the output',
+      () {
+        var userList = [user];
+        var stringifiedJson = SerializationManager.encodeForProtocol(userList);
+        expect(stringifiedJson.contains('password'), isFalse);
+      },
+    );
+
+    test(
+      'when a map containing user objects is encoded using encodeForProtocol method, then passwords are excluded from the output',
+      () {
+        var userMap = {
+          'user': user,
+          'users': [user]
+        };
+        var stringifiedJson = SerializationManager.encodeForProtocol(userMap);
+        expect(stringifiedJson.contains('password'), isFalse);
+      },
+    );
+  });
+
+  group('Given a list of primitives,', () {
+    var primitiveList = [1, 2.5, 'hello', true];
+
+    test(
+      'when encoded using encodeForProtocol method, then output matches the input list',
+      () {
+        var stringifiedJson =
+            SerializationManager.encodeForProtocol(primitiveList);
+        expect(stringifiedJson, '[1,2.5,"hello",true]');
+      },
+    );
+  });
+
+  group('Given a map of primitives,', () {
+    var primitiveMap = {'a': 1, 'b': 2.5, 'c': 'hello', 'd': true};
+
+    test(
+      'when encoded using encodeForProtocol method, then output matches the input map',
+      () {
+        var stringifiedJson =
+            SerializationManager.encodeForProtocol(primitiveMap);
+        expect(stringifiedJson, '{"a":1,"b":2.5,"c":"hello","d":true}');
+      },
+    );
+  });
+
+  group('Given a list of complex objects,', () {
+    late _User user;
+
+    setUp(() {
+      user = _User(name: 'John', password: '123');
+    });
+
+    test(
+      'when encoded using encodeForProtocol method, then passwords are excluded from the output',
+      () {
+        var list = [user, user];
+        var stringifiedJson = SerializationManager.encodeForProtocol(list);
+        expect(stringifiedJson.contains('password'), isFalse);
+      },
+    );
+  });
+
+  group('Given a map of complex objects,', () {
```

Comment:
Question: What is the difference between this test and the test on line 82?

Reply:
removed, no need for this.